src/m365/spfx/commands/project/project-github-workflow-add.ts

```diff
@@ -0,0 +1,203 @@
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        name: typeof args.options.name !== 'undefined',
+        branchName: typeof args.options.branchName !== 'undefined',
+        manuallyTrigger: !!args.options.manuallyTrigger,
+        loginMethod: typeof args.options.loginMethod !== 'undefined',
+        scope: typeof args.options.scope !== 'undefined',
+        skipFeatureDeployment: !!args.options.skipFeatureDeployment,
+        overwrite: !!args.options.overwrite
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-n, --name [name]'
+      },
+      {
+        option: '-b, --branchName [branchName]'
+      },
+      {
+        option: '-m, --manuallyTrigger'
+      },
+      {
+        option: '-l, --loginMethod [loginMethod]',
+        autocomplete: SpfxProjectGithubWorkflowAddCommand.loginMethod
+      },
+      {
+        option: '-s, --scope [scope]',
+        autocomplete: SpfxProjectGithubWorkflowAddCommand.scope
+      },
+      {
+        option: '-u, --siteUrl [siteUrl]'
+      },
+      {
+        option: '--skipFeatureDeployment'
+      },
+      {
+        option: '--overwrite'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+
+        if (args.options.scope && args.options.scope === 'sitecollection') {
+          if (!args.options.siteUrl) {
+            return `siteUrl option has to be defined when scope set to ${args.options.scope}`;
+          }
+
+          const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.siteUrl);
+          if (isValidSharePointUrl !== true) {
+            return isValidSharePointUrl;
+          }
+        }
+
+        if (args.options.loginMethod && SpfxProjectGithubWorkflowAddCommand.loginMethod.indexOf(args.options.loginMethod) < 0) {
+          return `${args.options.loginMethod} is not a valid login method. Allowed values are ${SpfxProjectGithubWorkflowAddCommand.loginMethod.join('|')}`;
+        }
+
+        if (args.options.scope && SpfxProjectGithubWorkflowAddCommand.scope.indexOf(args.options.scope) < 0) {
+          return `${args.options.scope} is not a valid scope. Allowed values are ${SpfxProjectGithubWorkflowAddCommand.scope.join('|')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    this.projectRootPath = this.getProjectRoot(process.cwd());
+    if (this.projectRootPath === null) {
+      throw new CommandError(`Couldn't find project root folder`, SpfxProjectGithubWorkflowAddCommand.ERROR_NO_PROJECT_ROOT_FOLDER);
+    }
+
+    const solution: string = path.join(this.projectRootPath as string, 'sharepoint', 'solution');
```

Comment:
This could be problematic if the project hasn't been built yet. I suggest that we look for other way to determine the project's name, like package.json or .yo-rc.json.

Reply:
changed to get name from `package.json`. thanks to that we don't need to validate if package is present and we do not need it👍